runtime/mount/python_lib/lplots/widgets/workflow.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from typing import Any, Literal, NotRequired
+
+from latch_cli.services.launch.launch_v2 import CompletedExecution, launch
+
+from lplots.widgets.text import w_text_output
+
+from . import _emit, _state, widget
+from .button import ButtonWidget, w_button
+
+workflow_widget_type: Literal["workflow"] = "workflow"
+
+
+class WorkflowWidgetState(_emit.WidgetState[workflow_widget_type, str]):
+    label: str
+    readonly: bool
+    wf_name: str
+    params: dict[str, Any]
+    version: str | None
+    execution_id: NotRequired[str]
+    completed_execution: NotRequired[CompletedExecution]
+
+
+@dataclass(frozen=True, kw_only=True)
+class WorkflowWidget(widget.BaseWidget):
+    _button: ButtonWidget
+    _state: WorkflowWidgetState
+
+    async def value(self) -> CompletedExecution | None:
+        if self._button.value:
+            wf_name = self._state.get("wf_name")
+            execution = launch(
+                wf_name=wf_name,
+                params=self._state.get("params"),
+                version=self._state.get("version"),
+            )
+            self._state["execution_id"] = execution.id
+            self._state["completed_execution"] = None
+
+            _state.submit_widget_state()
+            completed_execution = await execution.wait()
+            self._state["completed_execution"] = completed_execution
+
+            w_text_output(
```

Comment:
this seems like a bad idea
maybe let the consumer code show its own output if it wants to?

Reply:
at the very least set `key` on this